File: Roadmap/06 - RECURSIVIDAD/python/qv1ko.py
def func(number):
    print(number)

    number = number + 1

    if (number <= 100):
        func(number)

def fact(number):
    if (number <= 0):
        return 0
    elif (number == 1):
        return 1
    else:
        return number * fact(number - 1)

def fib(pos):
    if (pos <= 1):
        return 0
    elif (pos == 2 or pos == 3):
        return 1
    else:
        return fib(pos - 1) + fib(pos - 2)

func(0)

print("---")

print(fact(4))

print("---")

print(fib(5))

File: Roadmap/06 - RECURSIVIDAD/typescript/qv1ko.ts
func(0);

console.log("---");

console.log(fact(12));

console.log("---");

console.log(fib(12));

function func(num: number) {

    console.log(num);

    num = num + 1;

    if (num <= 100) {
        func(num);
    }

}

function fact(num: number): number {
    if (num <= 0) {
        return 0;
    } else if (num == 1) {
        return 1;
    } else {
        return num * fact(num - 1);
    }
}

function fib(pos: number): number {
    if (pos <= 1) {
        return 0;
    } else if (pos == 2 || pos == 3) {
        return 1;
    } else {
        return fib(pos - 1) + fib(pos - 2);
    }
}
